plugins/USBPrinting/USBPrinterOutputDeviceManager.py

```diff
@@ -128,8 +129,9 @@ def getInstance(cls, engine = None, script_engine = None):
         return USBPrinterOutputDeviceManager._instance
 
     def _getDefaultFirmwareName(self):
-        machine_instance = Application.getInstance().getMachineManager().getActiveMachineInstance()
-        machine_type = machine_instance.getMachineDefinition().getId()
+        machine_manager_model = MachineManagerModel()
```

Comment:
Initialising a MachineManager instance to get at a single setting value seems a bit wasteful (MachineManager comes at quite some overhead). Come to think of it I think you could also use ```Application.getInstance().getGlobalContainerStack().getProperty("machine_heated_bed", "value")``` to check the setting.

That way you don't have to introduce a single-use method to MachineManagerModel.